server/catgenome/src/main/java/com/epam/catgenome/manager/wig/BedGraphManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2016 EPAM Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.epam.catgenome.manager.wig;
+
+import com.epam.catgenome.constant.MessagesConstants;
+import com.epam.catgenome.entity.reference.Chromosome;
+import com.epam.catgenome.entity.track.Track;
+import com.epam.catgenome.entity.wig.Wig;
+import com.epam.catgenome.entity.wig.WigFile;
+import com.epam.catgenome.manager.TrackHelper;
+import com.epam.catgenome.manager.wig.reader.BedGraphCodec;
+import com.epam.catgenome.manager.wig.reader.BedGraphReader;
+import com.epam.catgenome.manager.wig.reader.BedGraphFeature;
+import com.epam.catgenome.util.IOHelper;
+import com.epam.catgenome.util.NgbFileUtils;
+import htsjdk.samtools.util.PeekableIterator;
+import htsjdk.tribble.index.IndexFactory;
+import htsjdk.tribble.index.interval.IntervalTreeIndex;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.epam.catgenome.component.MessageHelper.getMessage;
+
+/**
+ * Manages all work with BedGraph files.
+ * */
+@Service
```

Comment:
I think we don't need AbstractWigManager implementations as Spring beans. Task a look at VcfReader interface and it's implementations.